### ISSUE TITLE
Error: Distribution "altaera" is not installed

### ISSUE THREAD

**Issue by LargeLanguageModelExplorer** [latestissue/altaera-packages.github.io#3]
I encountered the following error message after installing the entire package. I have carefully followed each step the guide have given. I also tried reinstalling it, but it still shows the same error message.

**Comment by latestissue** [comment]
How about if you type in "proot-distro install altaera"? This might solve the issue. Unusual that the issue persists after reinstalling, I'll look into that

**Comment by LargeLanguageModelExplorer** [comment]
Welp, it might be because of my device CPU's fault. Two error message pops up when I entered your suggested solution.
"The distribution download URL is not defined for CPU architecture 'arm'. Exciting due to failure"

**Comment by latestissue** [comment]
Oh, yes. The distro tarball is intended for "aarch64", which in fact is arm, but 64-bit. I'll see if I can implement the 32-bit one and I will let you know if it's possible to get it working on your device

**Comment by latestissue** [comment]
Okay, so I've added the 32-bit distribution and from now on it should work for your architecture, though I'm not exactly sure if further dependencies have their 32-bit versions of packages. You'll just have to do the reinstall and find out for yourself because I don't own a 32-bit Android device to test it out. If you can, do let me know if it does work now!

**Comment by latestissue** [comment]
Okay, the problem is bigger than it appears. Now I can remember why exactly I chose Artix Linux over Arch Linux as the PRoot Distro distribution - it's the fact that OpenBlas requires GLIBC 2.38 which is only packaged for Artix at this moment. Yeah, this sort of made a mess and I will have to do some workaround in order for the 32-bit architecture to work with this.. I'll probably have this fixed by tomorrow, I'll post when I have it done. Sorry for your inconvenience.

**Comment by LargeLanguageModelExplorer** [comment]
Thanks to you, I was able to install altaera, but there is still a problem and you mentioned it. It crashed when I try to start altaera due to some OpenBlas problem. I hope you have a great day.

**Comment by latestissue** [comment]
Could you try and reinstall once again now? But make sure you choose to build your own KoboldCpp in the installation process when asked, so that you have it compiled for the 32-bit arm architecture. 

I'm not convinced it will run at all, though. I've never tried compiling and running KoboldCpp with a 32-bit CPU. But, maybe it will. I have removed OpenBlas from the dependency list for 32-bit, because it's sort of redundant since I don't think any model utilising it (llama-based) could run on less than 6-8GB of RAM (32-bit has to be 4GB at most).

**Comment by LargeLanguageModelExplorer** [comment]
You're right, it ain't running. I got a slightly different error. This screenshot should probably help you solve the issue I am now dealing with.![Screenshot_20240202_095646_Termux.jpg](https://github.com/latestissue/AltaeraAI/assets/106442244/18ad2bd6-97f9-4cc2-a70e-0dbdc11f3a54)

**Comment by latestissue** [comment]
Have you reinstalled with selecting "build your own KoboldCpp"?

**Comment by latestissue** [comment]
If so, and the issue persists, then you could try logging into proot-distro "aelogin" >> "cd kcpp-ae" >> "make", wait until compiled, then "exit" >>"ae". Perhaps you have to compile without OpenBlas, though I don't think that would prevent you from running it anyway

**Comment by latestissue** [comment]
Then, if you still won't be able to run it, I think you'd be better off getting help from the maintainer of KoboldCpp himself (LostRuins), either on his repository, or on the KoboldAI Discord channel (probably the fastest way)

**Comment by LargeLanguageModelExplorer** [comment]
I did reinstalled with "build your own koboldcpp". I just tried "make" and this shows up. I think its just impossible to run this on 32 bit cpus.![Screenshot_20240202_182004_Termux.jpg](https://github.com/latestissue/AltaeraAI/assets/106442244/58e28c20-c47f-4141-8e0c-3b9004170bdb)

**Comment by latestissue** [comment]
I have just included the option "Build your own KoboldCpp [No-Blas]" if you wanted to try it nevertheless, as a kind of last resort. But yeah, to me it does look like it's unable to run on 32-bit. You could still try to talk to the guys at KoboldAI, though.

-- There's a big chance it won't compile unless a fresh git clone was performed, so you can try the "Build your own KoboldCpp [No-Blas]" option to find out
